services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.tsx

```diff
@@ -216,6 +241,17 @@ export const RateDetails = ({
 
         draftSubmission.rateInfos = cleanedRateInfos
 
+        const addtlActuaryContacts: ActuaryContact[] = []
+        rateInfos.forEach((rateInfo) => {
+            rateInfo.addtlActuaryContacts.forEach((contact) => {
+                addtlActuaryContacts.push(contact)
+            })
+        })
+        draftSubmission.addtlActuaryContacts = addtlActuaryContacts
```

Comment:
There are more changes needed here. The way we saved `addtlActuaryContacts` using protobuffs will not work with the implementation of the UI here. The gist here is that, in the old way we took all `addtlActuaryContacts` from the package level and copied them to each rate. So now we need to do it correctly and save each `addtlActuaryContacts` to their respective rate in `toProtoBuffer`. 

Here is a list of things I found so far that need to be changed.

- Update `toProtoBuffer` to not save `addtlActuaryContacts` at the package level, instead `rateInfos` should save its own additional actuary contacts in the `addtlActuaryContacts` at the rate level.
- `RateDetails`, remove `addtlActuaryContacts` from the `draftSubmission`. (line 250)
- In the API remove `addtlActuaryContacts` from the HPP level and only use the rate level `addtlActuaryContacts`
    - In the API `resolverHelpers.ts`  update the `convertHealthPlanPackageRatesToDomain` function to not use the package level `addtlActuaryContacts`.
    - In `convertContractWithRatesToHPP.ts` the function `convertContractWithRatesToFormData` Remove `addtlActuaryContacts` from HPP level and add to rate level.

Reply:
I think with the current code you will end up duplicating all the contacts across all the rates.

Reply:
I would look around more in the API to make sure I didn't miss a spot where this happens.